### PR TITLE
Convert Bash functions from include_mount_options_functions to separate Jinja macros

### PR DESCRIPTION
**Pull request by j-ode**
#### Description:

- Converted an include file containing bash functions include_mount_options_functions to a jinja macro and replaced the inclusion of said file with macro invocation.

#### Rationale:

- #7348

### CODE REVIEW

**Comment by jan-cerny** [comment]
It's an elegant solution which would allow you to proceed quickly. But, when I see it, I think it's not friendly to the developers that will want to use it, because they would have to open the macro definition and understand the bash code there. I think we should break it down and create a separate jinja macro for each of the small functions.

**Review by jan-cerny — CHANGES_REQUESTED** [review]

**Comment by jan-cerny** [comment]
@j-ode Also, after you do the changes, please rebase on the latest master.

File: shared/macros-bash.jinja [hunk 1]
```diff
@@ -687,3 +687,63 @@ dpkg-query --show --showformat='${db:Status-Status}\n' "{{{ pkgname }}}" 2>/dev/
...
 {{%- endif -%}}
 {{%- endmacro -%}}
+
+{{#
+# _vfstype:     type of filesystem
+# _new_opt:     new mount point option
+# _filesystem:  filesystem of new mount point (used when adding new entry in fstab)
+# _type:        mount type of new mount point (used when adding new entry in fstab)
+#}}
+{{% macro bash_ensure_mount_option_for_vfstype(_vfstype, _new_opt, _filesystem, _type) -%}}
```
[the remark this answers is not in the document]
Comment (matejak) [reply shared/macros-bash.jinja:697]: Moreover, if you rename `new_opt` to `new_mount_options`, then there will be nothing to document in that docstring.

Also use this opportunity to go through old comments and check out whether they make sense. What is referred to as "filesystem" is correct wrt `/etc/fstab` manpage, but I think that many people can confuse this with the "type" parameter. I highly recommend to include an example of fstab lines that show where arguments end up where.

File: shared/templates/mount_option/bash.template [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
-
-include_mount_options_functions
-
 function perform_remediation {
 	# test "$mount_has_to_exist" != 'no'
 	if test "{{{ MOUNT_HAS_TO_EXIST }}}" != 'no'; then
-		assert_mount_point_in_fstab {{{ MOUNTPOINT }}} || { echo "Not remediating, because there is no record of {{{ MOUNTPOINT }}} in /etc/fstab" >&2; return 1; }
+		# This macro gets expanded to code that will return 1 if MOUNTPOINT is not in /etc/fstab;
+		# This is consistent with the behavior prior to converting this function to a jinja macro
+		{{{ bash_assert_mount_point_in_fstab( MOUNTPOINT ) }}}
```
[the remark this answers is not in the document]
Comment (matejak) [reply shared/templates/mount_option/bash.template:9]: You can use e.g. https://cryptic-cliffs-32040.herokuapp.com/ to learn about how the dashes (e.g. `{{{- bash_assert_mount_point_in_fstab( MOUNTPOINT ) -}}}`) that eat leading or trailing whitespace work.

**Comment by jan-cerny** [comment]
Thank you very much for your pull request.

**Pull request merged**